dts/bindings/charger/maxim,max20335-charger.yaml

```diff
@@ -8,12 +8,20 @@ include: battery.yaml
 compatible: "maxim,max20335-charger"
 
 properties:
-  constant-charge-current-max-microamp:
-    required: true
-
   constant-charge-voltage-max-microvolt:
     required: true
 
+  chgin-to-sys-current-limit-microamp:
```

Comment:
I would set it as a required property or use one of the enum values as the default one as this chip has a couple of variants with different default register values. Then you could drop the `PROP_NOT_USED` from the code.

Reply:
If you want to drop `PROP_NOT_USED` I would prefer to use all properites as required.
As you said, there is several chip variants which differs in default configuration.
It would be safer to set all configuration entries correctlly based on the battery spec.

Reply:
Modified in a way written above.

Reply:
LGTM! Please check if we need to add a note in the release-note about that.